Various tweaks to joins vignette (#6688)
Rdatatable/data.table: vignettes/datatable-joins.Rmd

```

> ##### Summarizing with on in `data.table`

Reviewer: ```suggestion
##### Summarizing with `on` in `data.table`
```

                nomatch = NULL]
```

> Despite both tables having the same information, they present some relevant differences:

Reviewer: ```suggestion
Despite both tables having the same information, there are some relevant differences:
```

"they present" is redundant with the first bullet point

- They present different column ordering.
- They have column name differences:

>   - The `id` column on the first table has the same information as the `product_id` in the second table.
>   - The `i.id` column on the first table has the same information as the `id` in the second table.

Reviewer: ```suggestion
  - The `id` column in the first table has the same information as the `product_id` in the second table.
  - The `i.id` column in the first table has the same information as the `id` in the second table.
```
